### PR TITLE
Add ignoreFunctions: [] to length-zero-no-unit

### LINKED ISSUE #4945: Add ignoreFunctions: [] to length-zero-no-unit
> Clearly describe the bug

The `length-zero-no-unit` rule `ignore: ["custom-properties"]` option will only run on custom property declarations like:

```css
a { --x: 0px; }
```

It doesn't handle var functions like:

```
font: var(--x, 0px);
```

I see it being discussed here: https://github.com/stylelint/stylelint/pull/4860#issuecomment-662540528 but don't see a corresponding new issue so I created one. Maybe this would be considered a new feature? 🤔

> Which rule, if any, is the bug related to?

`length-zero-no-unit`

> What code is needed to reproduce the bug?

```css
.foo {
  --x: 0px;
  margin-top: var(--y, 0px);
}
```

> What stylelint configuration is needed to reproduce the bug?

e.g.

```json
{
  "rules": {
    "length-zero-no-unit": [
      true,
       {
         "ignore": ["custom-properties"]
       }
    ]
  }
}
```

> Which version of stylelint are you using?

`13.6.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

VScode plugin with `settings.json`

```
    "stylelint.validate": ["scss"],
    "editor.codeActionsOnSave": {
        "source.fixAll.stylelint": true
    }
```

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

I expect the unit on `0px` within `var(--y, 0px);` to be ignored.

> What actually happened (e.g. what warnings or errors did you get)?

"The following warnings were flagged:"

```shell
   Expected unit (length-zero-no-unit)
```

### PR DESCRIPTION
**Pull request by jeddy3**
> Which issue, if any, is this issue related to?

Closes #5082
Closes #4945

> Is there anything in the PR that needs further explanation?

Implements what I wrote in https://github.com/stylelint/stylelint/pull/5082#issuecomment-826062935

### CODE REVIEW

**Review by mattxwang — APPROVED** [review]
A short and sweet change, LGTM!

**Review by ybiquitous — APPROVED** [review]
I've left some comments, but this change looks good. 👍🏼

File: lib/rules/length-zero-no-unit/__tests__/index.js [hunk 1]
```diff
@@ -600,6 +600,29 @@ testRule({
 	],
 });
 
+testRule({
+	ruleName,
+	config: [true, { ignoreFunctions: ['var', /^--/] }],
```
[the remark this answers is not in the document]
Comment (jeddy3) [reply lib/rules/length-zero-no-unit/__tests__/index.js:605]: Good catch.

Strange though, as this should of been caught by the jest preset:

https://github.com/stylelint/jest-preset-stylelint/blob/882f30a55f3ba74c5cae939d178458f8566a6455/getTestRule.js#L109-L114

**Comment by jeddy3** [comment]
> - Added: `ignoreFunctions: []` to `length-zero-no-unit` ([#5314](https://github.com/stylelint/stylelint/pull/5314)).

File: lib/rules/length-zero-no-unit/__tests__/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/rules/length-zero-no-unit/__tests__/index.js:605]: Um..., if `schema.fix` is falsy, it is returned immediately, so it seems to me that a test finishes successfully. 🤔 

https://github.com/stylelint/jest-preset-stylelint/blob/882f30a55f3ba74c5cae939d178458f8566a6455/getTestRule.js#L107

**Pull request merged**